### PR TITLE
MINOR: [CI][Java] Update develocity access key environment variable

### PR DESCRIPTION
**Pull request by laurentgo**
### Rationale for this change

It was not mentioned in the migration document but `GRADLE_ENTERPRISE_ACCESS_KEY` environment variable is being deprecated and replaced with `DEVELOCITY_ACCESS_KEY`

### What changes are included in this PR?

Changing github java workflows referencing the legacy variable to use the new environment variable

### Are these changes tested?

No

### Are there any user-facing changes?

No

### CODE REVIEW

**Review by assignUser — APPROVED** [review]
LGTM Is the new name already active or do we need to wait for a release/updated pom?

**Comment by vibhatha** [comment]
cc @lidavidm

**Comment by laurentgo** [comment]
> LGTM Is the new name already active or do we need to wait for a release/updated pom?

The new name is already active and is written to stdout during a maven build when using the deprecated variable as shown in https://github.com/apache/arrow/actions/runs/9294383801/job/25579432229#step:6:17529

**Comment by lidavidm** [comment]
@github-actions crossbow submit -g java

**Review by lidavidm — APPROVED** [review]

**Comment by vibhatha** [comment]
```bash
[WARNING] Unexpected error while contacting Develocity server at https://ge.apache.org/
```

I see this warning locally (tried it before merging this PR), is this related?

**Comment by assignUser** [comment]
That indeed also appears [on main](https://github.com/apache/arrow/actions/runs/9310718441/job/25628642597#step:6:3829) @laurentgo I don't see it on previous java committs.

**Comment by vibhatha** [comment]
Btw, the `mvn clean install` holds without returning for a little while even after build success and then this warning. Seems like it pings this server and that's why the hold.

**Comment by laurentgo** [comment]
It seems ge.apache.org may be half down? I used it a couple of days ago and was able to see the build stats for the project but now I only see a webpage saying that develocity is starting up.

**Comment by assignUser** [comment]
What ever it was it seems to be back up :)

**Pull request merged**
